File: code/apipython/email_.py
import smtplib
import constantes
from email.mime.text import MIMEText

class email(object):
    __usuario       = constantes._EMAIL_USER_
    __senha         = constantes._EMAIL_PASSWORD_
    __host          = constantes._EMAIL_HOST_
    __porta         = constantes._EMAIL_PORT_
    __protocolo     = constantes._EMAIL_PROTOCOL_
    __erro          = ''
    
    def temErro(self):
        return self.__erro != '' and self.__erro!= None

    def getErro(self):
        return self.__erro

    def enviar(self, para, assunto, texto):
        try:
            self.__erro = ''  
            server = smtplib.SMTP(self.__host, self.__porta)
            server.starttls()
            server.login(self.__usuario, self.__senha)
            msg = MIMEText(texto)
            msg['From'] = self.__usuario
            msg['To'] = para
            msg['Subject'] = assunto          
            server.sendmail(self.__usuario, para, msg.as_string())
            server.quit()
            return True
        except Exception as e:
            self.__erro = str(e)
            return False
        except :
            self.__erro = 'Erro Inexperado ao enviar email'
            return False
        

File: code/apinode/email.js
const constantes = require('./constantes');
var nodemailer = require('nodemailer');

class email {
    constructor()
    {
        this.usuario       = constantes._EMAIL_USER_;
        this.senha         = constantes._EMAIL_PASSWORD_;
        this.host          = constantes._EMAIL_HOST_;
        this.porta         = constantes._EMAIL_PORT_;
        this.protocolo     = constantes._EMAIL_PROTOCOL_;
        this.erro          = '';
    }

    
    temErro(){
        return this.erro != '' && this.erro!=null;
    }

    getErro(){
        return this.erro;
    }

    enviar(para, assunto, texto, async){
        try{
            var self = this;
            this.erro = ''  
            var email = {
                from: this.usuario, // Quem enviou este e-mail
                to: para, // Quem receberá
                subject: assunto,  // Um assunto bacana :-) 
                html: texto // O conteúdo do e-mail
            };
            var transporte = nodemailer.createTransport({
                host: this.host,
                port: this.porta,
//                ignoreTLS: false,
//                requireTLS: true,
//                secure: true, 
                service: 'Gmail',
                auth: {
                  user: this.usuario, // Basta dizer qual o nosso usuário
                  pass: this.senha    // e a senha da nossa conta
                } 
            });            
            transporte.sendMail(email, function(err){
                if(err)
                  self.erro=err; 
                async();
            });
        }catch(exception){
            this.erro = exception;
            async();
        }
    }
}

module.exports = email;
